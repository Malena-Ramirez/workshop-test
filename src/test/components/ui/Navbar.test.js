import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { MemoryRouter, Router } from 'react-router-dom';
import { Navbar } from '../../../components/ui/Navbar';

configure({ adapter: new Adapter() });

describe('Prueba con componente <Navbar/>', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: 'Lola'
    }
  }

  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn()
  }

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('Validar el contenido del nombre del usuario', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Lola');
  })

  test('Debe llamar la funciÃ³n logout y usar history', () => {
    wrapper.find('button').prop('onClick')();
    expect(historyMock.replace).toHaveBeenCalledWith('/login')
  })
})