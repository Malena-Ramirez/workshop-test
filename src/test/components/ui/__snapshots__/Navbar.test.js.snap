// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con componente <Navbar/> Validar el contenido del nombre del usuario 1`] = `ReactWrapper {}`;
